hw/ip/usbdev/data/usbdev_testplan.hjson

```diff
@@ -815,5 +815,57 @@
       stage: V2
       tests: ["usbdev_nak_to_out_trans_when_avbuffer_empty_rxfifo_full"]
     }
+    {
+      name: data_toggle_restore
+      desc: '''
+            Verify that each of the IN and OUT Data Toggle bits may be both set and cleared by
+            software to ensure that resuming communications after returning from Deep Sleep
+            is possible.
+
+            - Ensure that the USB device disconnected (pull up not enabled) such that no
+              communications shall occur with the device.
+            - Set all of the even-numbered OUT Data Toggle bits to 1 and all of the odd-numbered
+              OUT Data Toggle bits to 0.
+            - Check that the OUT Data Toggle bits read as 0x555.
+            - Set all of the even-numbered IN Data Toggle bits to 0 and all of the odd-numbered
+              IN Data Toggle bits to 1.
+            - Check that the IN Data Toggle bits read as 0xAAA.
+            - Check that the OUT Data Toggle bits read as 0x555.
+            - Invert all of the OUT Data Toggle bits.
+            - Invert all of the IN Data Toggle bits.
+            - Check that the OUT Data Toggle bits read as 0xAAA.
+            - Check that the IN Data Toggle bits read as 0x555.
+            - Set all of the OUT Data Toggle bits to random values.
+            - Set all of the IN Data Toggle bits to random values.
+            - Check that all of the OUT and IN Data Toggle bits read back as expected.
+            '''
+      stage: V2
+      tests: ["usbdev_data_toggle_restore"]
+    }
+    {
+      name: setup_priority
+      desc: '''
+            Verify that the final slot of the Rx FIFO is reserved for use by SETUP packets only.
+            The final slot of the Rx FIFO buffer shall never be assigned to an OUT DATA packet
+            because it could force the USB device to be unresponsive to a SETUP DATA packet.
+
+            - Configure the USB device to receive SETUP packets on Endpoint Zero.
+            - Configure endpoint 1 to receive OUT DATA packets.
+            - Place 1 buffer in the Available SETUP Buffer FIFO.
+            - Place 8 buffers in the Available OUT Buffer FIFO.
+            - Transmit 7 OUT DATA packets to endpoint 1, checking that each packet is
+              ACKed, but do not read from Rx FIFO.
+            - Check that the Rx FIFO depth contains 7 packets.
```

Comment:
Nit: The depth is a number and doesn't contain anything :-) If we want to mention the depth explicitly, maybe "Use the depth register to check that the Rx FIFO contains 7 packets." ?